upf/upf_pfcp_server.c

```diff
@@ -1040,11 +1040,30 @@ static void upf_validate_session_timers ()
 
 #endif
 
-void upf_pfcp_server_stop_timer (u32 handle)
+void upf_pfcp_server_stop_msg_timer (pfcp_msg_t * msg)
 {
+  if (msg->timer == ~0)
+    return;
+
   pfcp_server_main_t *psm = &pfcp_server_main;
+  u32 id_t1 = ((0x80 | PFCP_SERVER_T1) << 24) | msg->seq_no;
+  u32 id_resp =
+    ((0x80 | PFCP_SERVER_RESPONSE) << 24) | pfcp_msg_get_index (psm, msg);
+  u32 *exp_id;
```

Comment:
personal, I don't this style. I would first declare the variable, the check msg->timer and the initialize the variable.

```suggestion
  pfcp_server_main_t *psm = &pfcp_server_main;
  u32 id_resp, *exp_id, id_t1;

    if (msg->timer == ~0)
    return;

  id_t1 = ((0x80 | PFCP_SERVER_T1) << 24) | msg->seq_no;
  id_resp =
    ((0x80 | PFCP_SERVER_RESPONSE) << 24) | pfcp_msg_get_index (psm, msg);
```

Reply:
Okay, will fix. Did too much C++ and Go in my time :)

Reply:
Fixed